internal/smb2/request.go

```diff
@@ -1012,6 +1012,23 @@ func (r CancelRequestDecoder) StructureSize() uint16 {
 	return le.Uint16(r[:2])
 }
 
+// --------------------------------------------------------------------------
+//	REQ_GET_DFS_REFERRAL -- DFS Referral Request
+//
+type DFSReferralRequest struct {
+	MaxReferralLevel uint16
+	RequestFileName  string
+}
+
+func (d *DFSReferralRequest) Size() int {
+	return int(uint16(2) + uint16(utf16le.EncodedStringLen(d.RequestFileName)))
+}
+
+func (d *DFSReferralRequest) Encode(b []byte) {
+	le.PutUint16(b, d.MaxReferralLevel)
+	utf16le.EncodeString(b[2:], d.RequestFileName)
```

Comment:
**issue:** Compilation error: `undefined: utf16le.EncodeString`.

`utf16le.EncodeString()` and `utf16le.DecodeToString` are not included in this PR. My guess is you're missing a file.